compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -13717,8 +13716,7 @@ private STNode parseRetryTypeParamRhs(STNode retryKeyword, STNode typeParam) {
 
         STNode blockStmt = parseRetryBody();
         endContext(); // end retry-stmt
-        STNode onFailClause = parseOptionalOnFailClause();
-        return STNodeFactory.createRetryStatementNode(retryKeyword, typeParam, args, blockStmt, onFailClause);
+        return STNodeFactory.createRetryStatementNode(retryKeyword, typeParam, args, blockStmt);
```

Comment:
Let's remove the error handler path added for on-fail-clause these changes. 

https://github.com/ballerina-platform/ballerina-lang/blob/2d730a699ebe7ff6be930cfe7e47a2b09c38473c/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java#L4735

Reply:
I will remove it.

Reply:
Updated. 6e28d27522656155f4d8712f60a34356460eecca